File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/effect/UXGlowEffect.java
package org.develnext.jphp.ext.javafx.classes.effect;

import javafx.scene.effect.Bloom;
import javafx.scene.effect.Effect;
import javafx.scene.effect.Glow;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "effect\\UXGlowEffect")
public class UXGlowEffect extends UXEffect<Glow> {
    interface WrappedInterface {
        @Property double level();
        @Property Effect input();
    }

    public UXGlowEffect(Environment env, Glow wrappedObject) {
        super(env, wrappedObject);
    }

    public UXGlowEffect(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new Glow();
    }

    @Signature
    public void __construct(double level) {
        __wrappedObject = new Glow(level);
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/effect/UXGlowEffect.php
<?php
namespace php\gui\effect;

use php\gui\paint\UXColor;

/**
 * Class UXGlowEffect
 * @package php\gui\effect
 * @packages gui, javafx
 */
class UXGlowEffect extends UXEffect
{
    /**
     * @var double   0.0 to 1.0
     */
    public $level = 0.3;

    /**
     * @param double $level (optional)
     */
    public function __construct($level)
    {
    }
}